src/slices/BondSlice.ts

```diff
@@ -227,6 +228,14 @@ export const bondAsset = createAsyncThunk(
     } finally {
       if (bondTx) {
         segmentUA(uaData);
+        ReactGA.event({
+          category: "Bonds",
+          action: uaData.type ?? "unknown",
+          value: parseFloat(uaData.value),
+          label: uaData.bondName,
+          dimension1: uaData.txHash ?? "unknown",
+          dimension2: uaData.address,
+        });
```

Comment:
as a general comment on all of these events. Is there any chance that the ReactGA.event fails somewhere on GA's end & does not return? Because if so then the `dispatch(clearPendingTxn...` would not fire on the next line & the user's button state will be stuck in `Pending`.

Inside of `segmentUA` we wrap the call in a `try/catch` for that reason (I forget why but it failed on Segment's end in certain scenarios).

Reply:
@cptnanon found error handling within ReactGa. so we're all good.